### ISSUE TITLE
Error initializing PrismaClient on Windows Arm64

### ISSUE THREAD

**Issue by YanGidorini** [prisma/prisma#25206]
### Bug description

I am able to generate prisma client
```
PS C:\Users\yan-yoga\Documents\codar\portal-tela\libs\migrations> npm run build

> tela-migrations@1.0.0 build
> npx prisma generate --schema=./src/prisma/schema.prisma && tsc

Prisma schema loaded from src\prisma\schema.prisma

✔ Generated Prisma Client (v5.19.1) to .\..\..\node_modules\@prisma\client in 84ms

Start by importing your Prisma Client (See: http://pris.ly/d/importing-client)
```

```
import { PrismaClient } from '@prisma/client'

export const prisma = new PrismaClient()

await prisma.execution.create(
  {
    data: {
      id: command.data.execution.id,
      source: command.data.execution.scraper.name
    }
  })
  ...
```

but i can't use the generated client, because the following error is thrown

```
PrismaClientInitializationError: Unable to require(`C:\Users\yan-yoga\Documents\codar\portal-tela\node_modules\.prisma\client\query_engine-windows.dll.node`).
The Prisma engines do not seem to be compatible with your system. Please refer to the documentation about Prisma's system requirements: https://pris.ly/d/system-requirements

Details: \\?\C:\Users\yan-yoga\Documents\codar\portal-tela\node_modules\.prisma\client\query_engine-windows.dll.node is not a valid Win32 application.       
\\?\C:\Users\yan-yoga\Documents\codar\portal-tela\node_modules\.prisma\client\query_engine-windows.dll.node
    at Object.loadLibrary (C:\Users\yan-yoga\Documents\codar\portal-tela\node_modules\@prisma\client\runtime\library.js:110:10219)
    at process.processTicksAndRejections (node:internal/process/task_queues:95:5)
    at async Cr.loadEngine (C:\Users\yan-yoga\Documents\codar\portal-tela\node_modules\@prisma\client\runtime\library.js:111:448)
    at async Cr.instantiateLibrary (C:\Users\yan-yoga\Documents\codar\portal-tela\node_modules\@prisma\client\runtime\library.js:110:12575) {
  clientVersion: '5.19.1',
  errorCode: undefined
}
```

### How to reproduce

use prisma on windows 11 arm

### Expected behavior

the query engine does not throw any error

### Prisma information

```prisma
generator client {
  provider = "prisma-client-js"
}

datasource db {
  provider = "postgresql"
  url      = env("DATABASE_URL")
}

model Execution {
  id           String       @id @db.Uuid
  insertedDate DateTime     @default(now()) @db.Timestamptz
  source       String?      @db.VarChar(64)
  read         Status       @default(Pending)
}

enum Status {
  Pending
  InProgress
  Completed
  CompletedWithDelay
  Flagged
  WatermarkDetected
  Delayed
  Duplicated
  MergedPending
  MergedCompleted
  MergedFlagged
}
```

```ts
import { PrismaClient } from '@prisma/client'

export const prisma = new PrismaClient()

await prisma.execution.create(
  {
    data: {
      id: command.data.execution.id,
      source: command.data.execution.scraper.name
    }
  })
  ...
```

### Environment & setup

- **OS:** Windows 11 Home, ARM64
- **Database:** PostgreSQL
- **Node.js version:** v20.17.0 (ARM64 binaries)
- **Device:** Lenovo Yoga 7x Gen 9

## Prisma Version

```
PS C:\Users\yan-yoga\Documents\codar\portal-tela> prisma -v
Error: Unable to require(`C:\Users\yan-yoga\AppData\Roaming\npm\node_modules\prisma\node_modules\@prisma\engines\query_engine-windows.dll.node`).
The Prisma engines do not seem to be compatible with your system. Please refer to the documentation about Prisma's system requirements: https://pris.ly/d/system-requirements

Details: \\?\C:\Users\yan-yoga\AppData\Roaming\npm\node_modules\prisma\node_modules\@prisma\engines\query_engine-windows.dll.node is not a valid Win32 application.
\\?\C:\Users\yan-yoga\AppData\Roaming\npm\node_modules\prisma\node_modules\@prisma\engines\query_engine-windows.dll.node
    at pne (C:\Users\yan-yoga\AppData\Roaming\npm\node_modules\prisma\build\index.js:284:116)
    at lWe (C:\Users\yan-yoga\AppData\Roaming\npm\node_modules\prisma\build\index.js:284:269)
prisma                  : 5.19.1
@prisma/client          : 5.19.1
Computed binaryTarget   : windows
Operating System        : win32
Architecture            : arm64
Node.js                 : v20.17.0
Query Engine (Node-API) : E_CANNOT_RESOLVE_VERSION (at ..\..\..\AppData\Roaming\npm\node_modules\prisma\node_modules\@prisma\engines\query_engine-windows.dll.node)
Schema Engine           : schema-engine-cli 69d742ee20b815d88e17e54db4a2a7a3b30324e3 (at ..\..\..\AppData\Roaming\npm\node_modules\prisma\node_modules\@prisma\engines\schema-engine-windows.exe)
Schema Wasm             : @prisma/prisma-schema-wasm 5.19.1-2.69d742ee20b815d88e17e54db4a2a7a3b30324e3
Default Engines Hash    : 69d742ee20b815d88e17e54db4a2a7a3b30324e3
Studio                  : 0.502.0
```

**Comment by aqrln** [comment]
Is your Node.js natively compiled for ARM64 or are you running the x86_64 version in emulation? If it's a native ARM build, then I wouldn't expect it to be able to load the Prisma query engine which is only compiled for x86_64 on Windows (Rosetta on macOS is not able to mix different architectures in a single process, I assume the same goes for Microsoft's binary translator).

As a workaround for now, you can either use an x86_64 version of Node.js, or build Prisma engines from sources.

**Comment by aqrln** [comment]
Another workaround you could use is either adding `engineType = "binary"` to the generator block in your schema and re-generating the client, or, better yet, running both `prisma generate` and your app with `PRISMA_CLIENT_ENGINE_TYPE=binary` environment variable set so you don't force the less performant binary engine to be used outside of your local environment. Since the query engine will be a separate process this way, Windows should be able to run it under binary translation even if Node.js is a native ARM64 binary.

**Comment by jponsano** [comment]
> Another workaround you could use is either adding `engineType = "binary"` to the generator block in your schema and re-generating the client, or, better yet, running both `prisma generate` and your app with `PRISMA_CLIENT_ENGINE_TYPE=binary` environment variable set so you don't force the less performant binary engine to be used outside of your local environment. Since the query engine will be a separate process this way, Windows should be able to run it under binary translation even if Node.js is a native ARM64 binary.

This is perfect and worked for me.  I had previously installed Node.js x86_64 to get around the issue, but was able to replace it iwth the ARM64 build and use `PRISMA_CLIENT_ENGINE_TYPE=binary`.

**Comment by aqrln** [comment]
I'm going to close this as a duplicate of https://github.com/prisma/prisma/issues/15306, please upvote that issue as we take the upvotes into account when prioritizing.

**Comment by SeuHkx** [comment]
> This is perfect and worked for me. I had previously installed Node.js x86_64 to get around the issue, but was able to replace it iwth the ARM64 build and use `PRISMA_CLIENT_ENGINE_TYPE=binary`.

thanks！！！

**Comment by cryskram** [comment]
> Another workaround you could use is either adding `engineType = "binary"` to the generator block in your schema and re-generating the client, or, better yet, running both `prisma generate` and your app with `PRISMA_CLIENT_ENGINE_TYPE=binary` environment variable set so you don't force the less performant binary engine to be used outside of your local environment. Since the query engine will be a separate process this way, Windows should be able to run it under binary translation even if Node.js is a native ARM64 binary.

this worked!

**Comment by tinncdev** [comment]
Stepped on this issue today trying to test windows env in parallel vm (apple silicon m3)
The engine type approach above didn't work for me but the new [queryCompiler](https://www.prisma.io/docs/orm/prisma-client/setup-and-configuration/no-rust-engine) does, and also look simpler than tinkering with environment variables & platfom though
Also, the windows on arm support status should be mentioned in the [System requirements](https://www.prisma.io/docs/orm/reference/system-requirements)
